src/objects.rs

```diff
@@ -1048,6 +1090,110 @@ impl GetObjectType for StopPoint {
     }
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub struct StopLocation {
+    pub id: String,
+    pub name: String,
+    #[serde(skip)]
+    pub comment_links: CommentLinksT,
+    pub visible: bool,
+    pub coord: Coord,
+    pub parent_id: Option<String>,
+    pub timezone: Option<String>,
+    pub geometry_id: Option<String>,
+    pub equipment_id: Option<String>,
+    pub level_id: Option<String>,
+    #[serde(skip)]
+    pub stop_type: StopType,
+}
+impl_id!(StopLocation);
+impl_comment_links!(StopLocation);
+
+impl AddPrefix for StopLocation {
+    fn add_prefix(&mut self, prefix: &str) {
+        self.id = prefix.to_string() + &self.id;
+        self.parent_id = self.parent_id.as_ref().map(|id| prefix.to_string() + &id);
+        self.geometry_id = self.geometry_id.as_ref().map(|id| prefix.to_string() + &id);
+        self.equipment_id = self
+            .equipment_id
+            .as_ref()
+            .map(|id| prefix.to_string() + &id);
+        self.level_id = self.level_id.as_ref().map(|id| prefix.to_string() + &id);
+    }
+}
+
+#[derivative(Default)]
+#[derive(Derivative, Serialize, Deserialize, Debug, Clone, PartialEq, Eq, Hash)]
+pub enum PathwayMode {
+    #[derivative(Default)]
+    #[serde(rename = "1")]
+    Walkway,
+    #[serde(rename = "2")]
+    Stairs,
+    #[serde(rename = "3")]
+    MovingSidewalk,
+    #[serde(rename = "4")]
+    Escalator,
+    #[serde(rename = "5")]
+    Elevator,
+    #[serde(rename = "6")]
+    FareGate,
+    #[serde(rename = "7")]
+    ExitGate,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Default, Clone)]
+pub struct Pathway {
+    #[serde(rename = "pathway_id")]
+    pub id: String,
+    pub from_stop_id: String,
+    #[serde(skip)]
+    pub from_stop_type: StopType,
+    pub to_stop_id: String,
+    #[serde(skip)]
+    pub to_stop_type: StopType,
+    pub pathway_mode: PathwayMode,
+    #[serde(
+        deserialize_with = "de_from_u8_with_check",
+        serialize_with = "ser_from_bool"
+    )]
+    pub is_bidirectional: bool,
+    #[serde(default, deserialize_with = "de_option_positive_decimal")]
+    pub length: Option<Decimal>,
+    pub traversal_time: Option<u32>,
+    #[serde(default, deserialize_with = "de_option_non_null_integer")]
+    pub stair_count: Option<i16>,
```

Comment:
Hmm, can we have a negative number of stairs?
```suggestion
    pub stair_count: Option<u16>,
```

Reply:
A positive stair_count implies that the rider walk up from from_stop_id to to_stop_id. And a negative stair_count implies that the rider walk down from from_stop_id to to_stop_id.

Reply:
Sorry for this one, I missed that in the specification.